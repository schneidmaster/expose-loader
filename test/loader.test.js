import webpack from './helpers/compiler';

describe('Loader', () => {
  test('exposes module with webpack configuration syntax', async () => {
    const config = {
      loader: {
        test: /foo\.js$/,
        options: 'foo',
      },
    };

    const stats = await webpack('webpack.js', config);
    const { modules } = stats.toJson();
    const source = modules
      .filter((module) => module.name.includes('-exposed'))
      .map((module) => module.source);
    expect(source).toMatchSnapshot();
  });

  test('exposes module with inline import syntax', async () => {
    const stats = await webpack('inline.js', {});
    const { modules } = stats.toJson();
    const source = modules
      .filter((module) => module.name.includes('-exposed'))
      .map((module) => module.source);
    expect(source).toMatchSnapshot();
  });
});
