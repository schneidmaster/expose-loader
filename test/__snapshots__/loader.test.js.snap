// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader exposes module with inline import syntax 1`] = `
Array [
  "module.exports = global[\\"foo\\"] = require(\\"-!./foo.js\\");",
]
`;

exports[`Loader exposes module with webpack configuration syntax 1`] = `
Array [
  "module.exports = global[\\"foo\\"] = require(\\"-!./foo.js\\");",
]
`;
